### ISSUE TITLE
Enable the content property in project.json to libraries as well

### ISSUE THREAD

**Issue by livarcocc** [dotnet/cli#753]
Currently, we only honor the Content property in project.json for projects with the emitEntryPoint flag set to true. We should enable that for libraries as well.

Some test libraries need that and the System.Net.Libuv library in corefx needs that as well.

**Comment by livarcocc** [comment]
@davidfowl @piotrpMSFT Opinions?

**Comment by piotrpMSFT** [comment]
Can you add a scenario this unblocks? Seems ok in general, but I want to be sure we're encouraging good practices.

**Comment by andschwa** [comment]
Well this explains why my dependencies `content` gets completely ignored. Piggy-backing: for the top most dependency (the one with `emitEntryPoint` set to true), on Ubuntu 14.04, publishing content fails:

```
The process cannot access the file '../foo/bar' because it is being used by another process.
```

@piotrpMSFT this blocks our project from getting necessary files published in a cross platform manner. Right now we have duplicate code in our Windows and Linux build scripts to deploy the files , we want to use `content` or `contentFiles` in our main library to deploy sets of necessary files on publish.

**Comment by livarcocc** [comment]
@andschwa, I have a fix for the issue you are seeing regarding file being used by another process. It is currently in PR https://github.com/dotnet/cli/pull/1194. I am waiting for CI to pass to merge.

**Comment by andschwa** [comment]
@livarcocc That'll be great! I'll at least get the content deployment moved into the top most dependency and out of the build scripts, and then wait on this issue to move it to the library they belong to. Thanks :)

**Comment by andschwa** [comment]
@piotrpMSFT Another scenario this unblocks: being able to use a test runner. When a test runner (like xunit) takes a dependency on the code that it is testing, the required content does not get published, and so the tests fail for no good reason.

**Comment by andschwa** [comment]
Which actually means that the `emitEntryPoint` is not, in fact, the trigger. My tests have a dependency on a project that has it set to true, but building the tests does not cause the content to be published.

**Comment by livarcocc** [comment]
We don't have P2P dependency content copying in place (or project to package for that matter). @pakrym is working on it and apparently is getting closer for project to package.

**Comment by andschwa** [comment]
Thanks @livarcocc. With that being the case, I can add another work-around to my test wrapper script, to manually copy the content that needs to be published, with the goal of removing this once transitive content dependencies are fixed.

**Comment by piotrpMSFT** [comment]
The team is actively working on enabling [MSBuild](https://blogs.msdn.microsoft.com/dotnet/2016/05/23/changes-to-project-json/) and the component affected will be superseded by the new project system, so I am closing this issue.
